ci: Use same `merge_script` implementation for Windows as for all

hebasto opened the pull request:
This PR is a continuation of bitcoin/bitcoin#26202 and it suggests the same approach for the "Win64 native" CI task.

hebasto committed:
ci: Move `git config` commands into script where they are used

hebasto committed:
ci: Use same `merge_script` implementation for Windows as for all

MarcoFalke commented on ci/lint/06_script.sh:
```diff
@@ -31,6 +31,10 @@ if [ "$CIRRUS_REPO_FULL_NAME" = "bitcoin/bitcoin" ] && [ "$CIRRUS_PR" = "" ] ; t
     git log HEAD~10 -1 --format='%H' > ./contrib/verify-commits/trusted-sha512-root-commit
     git log HEAD~10 -1 --format='%H' > ./contrib/verify-commits/trusted-git-root
     mapfile -t KEYS < contrib/verify-commits/trusted-keys
+    if [ "$CI" = "true" ]; then
```
maybe remove this check, since there is already the `[ "$CIRRUS_REPO_FULL_NAME" = "bitcoin/bitcoin" ]` check?

hebasto replied:
Thanks! [Updated](https://github.com/bitcoin/bitcoin/pull/26236#issuecomment-1266546245).

MarcoFalke commented on ci/lint/06_script.sh:
```diff
@@ -31,6 +31,10 @@ if [ "$CIRRUS_REPO_FULL_NAME" = "bitcoin/bitcoin" ] && [ "$CIRRUS_PR" = "" ] ; t
     mapfile -t KEYS < contrib/verify-commits/trusted-keys
+    if [ "$CI" = "true" ]; then
+      git config --global user.email "ci@ci.ci"
+      git config --global user.name "ci"
```
maybe remove `--global`?

hebasto replied:
Thanks! [Updated](https://github.com/bitcoin/bitcoin/pull/26236#issuecomment-1266546245).

hebasto commented:
Updated 3d9a2e9af000ca7fcac3244677d5f50a3d6da562 -> 37cf4720635b63cbe36a900a2411718704b63899 ([pr26236.01](https://github.com/hebasto/bitcoin/commits/pr26236.01) -> [pr26236.02](https://github.com/hebasto/bitcoin/commits/pr26236.02), [diff](https://github.com/hebasto/bitcoin/compare/pr26236.01..pr26236.02)):

- addressed @MarcoFalke's comments

maflcko reviewed (approved):
lgtm

MarcoFalke commented on .cirrus.yml:
```diff
@@ -117,14 +114,7 @@ task:
-    # files are considered "modified" now. It will break the following `git merge`
-    # command. The next two commands make git ignore this issue.
-    - git config core.filemode false
-    - git reset --hard
```
From the red CI, looks like this is still required

hebasto replied:
Amended in #26249.
